src/Persistent/Idiom.hs

```diff
@@ -136,3 +140,24 @@ getUserViews = makeUserView <$$> getAllUsers
 
 findActiveUser :: AUID User -> MQuery User
 findActiveUser uid = (((^? activeUser) . makeUserView) =<<) <$> findUser uid
+
+-- This operation is idempotent, that is, freezing a frozen state
+-- has no effect and similarly for thawing. Otherwise we'd need to make
+-- sure the form treats vacuus setting of the freeze state as invalid.
+--
+-- Note that this operation is atomic, thus ensuring that all topics
+-- are frozen or all are not frozen. Problem could arise, e.g., when
+-- two admins concurrently freeze and thaw, with different speeds.
+saveAndEnactFreeze :: Timestamp -> Freeze -> AUpdate ()
+saveAndEnactFreeze now shouldBeFrozenOrNot = do
+  dbFreeze .= shouldBeFrozenOrNot
+  let change | shouldBeFrozenOrNot == Frozen = PhaseFreeze now
+             | otherwise = PhaseThaw now
```

Comment:
I will change this to a `case` expression: easier to read (IMO anyway), and you can make a list of all cases and get a compiler warning if you introduce a new constructor.